### PR TITLE
Add support for `@coqbot inline-stdlib=yes resume ci minimize ci-foo`

### PR DESCRIPTION
**Pull request by JasonGross**
Also for `@coqbot extra-arg=--inline-coqlib resume ci minimize ci-foo` and `@coqbot extra-arg=--inline-coqlib ci minimize ci-foo`

On top of #257 to avoid conflicts

For https://github.com/coq/coq/pull/16967#issuecomment-1372540099, in conjunction with https://github.com/JasonGross/coq-tools/pull/144

### CODE REVIEW

**Comment by JasonGross** [comment]
Seems to fail with `Error: Unhandled exception: (Invalid_argument Str.matched_group)`, not sure what's going on

File: src/helpers.ml [hunk 1]
```diff
@@ -2,12 +2,24 @@ open Base
...
   with Stdlib.Not_found -> false
 
+let rec fold_string_matches ~regexp ~f ~init ?(pos = 0) string =
+  if string_match ~regexp ~pos string then
+    let pos = Str.match_end () in
+    f (fun () -> fold_string_matches ~regexp ~f ~init ~pos string)
+  else init
+
+let map_string_matches ~regexp ~f string =
+  fold_string_matches ~regexp ~f:(fun rest -> f () :: rest ()) ~init:[] string
```
Comment (SkySkimmer) [inline src/helpers.ml:18]: execution order is not specified and usually ends up right-to-left so this is wrong
instead do
```suggestion
  fold_string_matches ~regexp ~f:(fun rest -> let v = f () in v :: rest ()) ~init:[] string
```

**Comment by JasonGross** [comment]
This seems to be working [here](https://github.com/coq/coq/pull/16977#issuecomment-1402868303) (well, the coqbot side of it at least; the Coq CI artifact side still has issues at https://github.com/coq/coq/issues/17159).  Shall we merge?

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
LGTM merge as you wish

**Pull request merged**